src/averageAges.js

```diff
@@ -14,13 +14,21 @@
  *
  * @return {number}
  */
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
-}
+  const men = century
+    ? people.filter(person => person.sex === 'm'
+    && Math.ceil(person.died / 100) === century)
+    : people.filter(person => person.sex === 'm');
+
+  const menAges = men.map(person => person.died - person.born);
+
+  const menAverageAge = menAges
+    .reduce((firstMan, secondMan) =>
+      (firstMan + secondMan)) / menAges.length;
+
+  return menAverageAge;
+};
```

Comment:
you duplicate this logic in each function, its better to move it to one separate function